apps/cli/lib/parser/mana_parser.ex

```diff
@@ -0,0 +1,155 @@
+defmodule CLI.Parser.ManaParser do
+  @moduledoc """
+  Parses for command line arguments for the CLI.
+  """
+
+  @default_chain_name "ropsten"
+  @default_no_discovery false
+  @default_no_sync false
+  @default_bootnodes "from_chain"
+  @default_debug false
+
+  @doc """
+  Parsers args for syncing
+
+  ## Options:
+    * `--chain` - Chain to load data from (default: ropsten)
+    * `--no-discovery` - Perform discovery (default: false)
+    * `--no-sync` - Perform syncing (default: false)
+    * `--bootnodes` - Comma separated list of bootnodes (default: from_chain)
+    * `--debug` - Add remote debugging (default: false)
+
+  ## Examples
+
+      iex> CLI.Parser.ManaParser.mana_args(["--chain", "ropsten", "--no-discovery", "--no-sync"])
+      {:ok, %{
+        chain_name: :ropsten,
+        discovery: false,
+        sync: false,
+        bootnodes: :from_chain,
+        debug: false
+      }}
+
+      iex> CLI.Parser.ManaParser.mana_args(["--chain", "foundation", "--bootnodes", "enode://google.com,enode://apple.com", "--debug"])
+      {:ok, %{
+        chain_name: :foundation,
+        discovery: true,
+        sync: true,
+        bootnodes: ["enode://google.com", "enode://apple.com"],
+        debug: true
+      }}
+
+      iex> CLI.Parser.ManaParser.mana_args([])
+      {:ok, %{
+        chain_name: :ropsten,
+        discovery: true,
+        sync: true,
+        bootnodes: :from_chain,
+        debug: false
+      }}
+
+      iex> CLI.Parser.ManaParser.mana_args(["--chain", "pony"])
+      {:error, "Invalid chain: pony"}
+  """
+  @spec mana_args([String.t()]) ::
+          {:ok,
+           %{
+             chain_name: atom(),
+             discovery: boolean(),
+             sync: boolean(),
+             bootnodes: :from_chain | list(String.t()),
+             debug: boolean()
+           }}
+          | {:error, String.t()}
+  def mana_args(args) do
+    {kw_args, _extra} =
+      OptionParser.parse!(args,
+        switches: [
+          chain: :string,
+          no_discovery: :boolean,
+          no_sync: :boolean,
+          bootnodes: :string,
+          debug: :boolean
+        ]
+      )
+
+    with {:ok, chain_name} <- get_chain_name(kw_args) do
+      with {:ok, discovery} <- get_discovery(kw_args) do
+        with {:ok, sync} <- get_sync(kw_args) do
+          with {:ok, bootnodes} <- get_bootnodes(kw_args) do
+            with {:ok, debug} <- get_debug(kw_args) do
+              {:ok,
```

Comment:
there's no need to nest with's this way, one is enough if statements are separated with commas

Reply:
yeah, that's way better, thanks.